### ISSUE TITLE
[Issue]: stopping trickplay task does not kill ffmpeg process

### ISSUE THREAD

**Issue by NotSaifA** [jellyfin/jellyfin#11787]
### Please describe your bug

When I cancel the 'Generate Trickplay Images' task, the expected behavior is to stop generating images. However, the observed behavior is that the ffmpeg process keeps running and generating images, even after the ffmpeg process exits and generates all the images they are completely ignored and will stay in the temp folder until it gets cleaned later.
When the task is started again it will start a new ffmpeg process for the same episode even though the previous process successfully exited and generated all the required images.

### Reproduction Steps

1. Start 'Generate Trickplay Images' task
2. Stop 'Generate Trickplay Images' task
3. Running top shows the ffmpeg process still running.
4. The ffmpeg process keeps running until it generates all images and saves them to the temp directory.
5. Starting the task again starts a new ffmpeg process for the same episode.
6. The images generated by the processes that were supposed to be killed are not used, and stay until cache gets cleaned.

### Jellyfin Version

10.9.0

### if other:

10.9.2

### Environment

```markdown
- OS: Ubuntu 22.04.4 LTS
- Linux Kernel: 5.15.0-107-generic
- Virtualization: Docker version 26.1.3
- Clients: n/a
- Browser: n/a
- FFmpeg Version: 6.0.1-Jellyfin
- Playback Method: n/a
- Hardware Acceleration: none
- GPU Model: none
- Plugins: none
- Reverse Proxy: none
- Base URL: none
- Networking: Bridge/NAT
- Storage: local
```


### Jellyfin logs

```shell
[2024-05-22 00:14:39.194 -04:00] [DBG] [24] Emby.Server.Implementations.ScheduledTasks.TaskManager: Executing "Generate Trickplay Images"
[2024-05-22 00:14:39.215 -04:00] [DBG] [24] Jellyfin.Server.Implementations.Trickplay.TrickplayManager: Trickplay refresh for 81beb50e-99bb-d24f-f059-2d4723b3271a (replace existing: False)
[2024-05-22 00:14:39.215 -04:00] [INF] [24] Jellyfin.Server.Implementations.Trickplay.TrickplayManager: Creating trickplay files at 320 width, for "/data/media/tv/show/Season 01/S01E01.mkv" [ID: 81beb50e-99bb-d24f-f059-2d4723b3271a]
[2024-05-22 00:14:39.216 -04:00] [INF] [24] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Trickplay generation: "/usr/lib/jellyfin-ffmpeg/ffmpeg -loglevel error -threads 16 -noautorotate -i file:\"/data/media/tv/show/Season 01/S01E01.mkv\" -an -sn -vf \"fps=0.10000000149011612,setparams=color_primaries=bt709:color_trc=bt709:colorspace=bt709,yadif=0:-1:0,scale=trunc(min(max(iw\,ih*a)\,320)/2)*2:trunc(ow/a/2)*2,format=yuv420p\" -threads 16 -c:v mjpeg -qscale:v 4 -f image2 \"/cache/temp/c0f25652dd4e41948578c18d999a2995/%08d.jpg\""
[2024-05-22 00:14:40.860 -04:00] [INF] [19] Emby.Server.Implementations.ScheduledTasks.TaskManager: Attempting to cancel Scheduled Task "Generate Trickplay Images"
[2024-05-22 00:14:59.218 -04:00] [ERR] [20] Jellyfin.Server.Implementations.Trickplay.TrickplayManager: Error creating trickplay images.
System.OperationCanceledException: The operation was canceled.
   at System.Threading.CancellationToken.ThrowOperationCanceledException()
   at System.Threading.CancellationToken.ThrowIfCancellationRequested()
   at MediaBrowser.MediaEncoding.Encoder.MediaEncoder.ExtractVideoImagesOnIntervalInternal(String inputArg, String filterParam, String vidEncoder, Nullable`1 outputThreads, Nullable`1 qualityScale, Nullable`1 priority, CancellationToken cancellationToken)
   at Jellyfin.Server.Implementations.Trickplay.TrickplayManager.RefreshTrickplayDataInternal(Video video, Boolean replace, Int32 width, TrickplayOptions options, CancellationToken cancellationToken)
[2024-05-22 00:14:59.219 -04:00] [INF] [20] Emby.Server.Implementations.ScheduledTasks.TaskManager: "Generate Trickplay Images" Cancelled after 0 minute(s) and 20 seconds

[2024-05-22 00:15:01.758 -04:00] [DBG] [42] Emby.Server.Implementations.ScheduledTasks.TaskManager: Executing "Generate Trickplay Images"
[2024-05-22 00:15:01.795 -04:00] [DBG] [42] Jellyfin.Server.Implementations.Trickplay.TrickplayManager: Trickplay refresh for 81beb50e-99bb-d24f-f059-2d4723b3271a (replace existing: False)
[2024-05-22 00:15:01.795 -04:00] [INF] [42] Jellyfin.Server.Implementations.Trickplay.TrickplayManager: Creating trickplay files at 320 width, for "/data/media/tv/show/Season 01/S01E01.mkv" [ID: 81beb50e-99bb-d24f-f059-2d4723b3271a]
[2024-05-22 00:15:01.796 -04:00] [INF] [42] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Trickplay generation: "/usr/lib/jellyfin-ffmpeg/ffmpeg -loglevel error -threads 16 -noautorotate -i file:\"/data/media/tv/show/Season 01/S01E01.mkv\" -an -sn -vf \"fps=0.10000000149011612,setparams=color_primaries=bt709:color_trc=bt709:colorspace=bt709,yadif=0:-1:0,scale=trunc(min(max(iw\,ih*a)\,320)/2)*2:trunc(ow/a/2)*2,format=yuv420p\" -threads 16 -c:v mjpeg -qscale:v 4 -f image2 \"/cache/temp/c9de8a33803048ef98f68d07a80f17a2/%08d.jpg\""
[2024-05-22 00:15:05.642 -04:00] [INF] [20] Emby.Server.Implementations.ScheduledTasks.TaskManager: Attempting to cancel Scheduled Task "Generate Trickplay Images"
[2024-05-22 00:15:21.808 -04:00] [ERR] [42] Jellyfin.Server.Implementations.Trickplay.TrickplayManager: Error creating trickplay images.
System.OperationCanceledException: The operation was canceled.
   at System.Threading.CancellationToken.ThrowOperationCanceledException()
   at System.Threading.CancellationToken.ThrowIfCancellationRequested()
   at MediaBrowser.MediaEncoding.Encoder.MediaEncoder.ExtractVideoImagesOnIntervalInternal(String inputArg, String filterParam, String vidEncoder, Nullable`1 outputThreads, Nullable`1 qualityScale, Nullable`1 priority, CancellationToken cancellationToken)
   at Jellyfin.Server.Implementations.Trickplay.TrickplayManager.RefreshTrickplayDataInternal(Video video, Boolean replace, Int32 width, TrickplayOptions options, CancellationToken cancellationToken)
[2024-05-22 00:15:21.808 -04:00] [INF] [42] Emby.Server.Implementations.ScheduledTasks.TaskManager: "Generate Trickplay Images" Cancelled after 0 minute(s) and 20 seconds

[2024-05-22 00:15:25.046 -04:00] [DBG] [20] Emby.Server.Implementations.ScheduledTasks.TaskManager: Executing "Generate Trickplay Images"
[2024-05-22 00:15:25.085 -04:00] [DBG] [20] Jellyfin.Server.Implementations.Trickplay.TrickplayManager: Trickplay refresh for 81beb50e-99bb-d24f-f059-2d4723b3271a (replace existing: False)
[2024-05-22 00:15:25.086 -04:00] [INF] [20] Jellyfin.Server.Implementations.Trickplay.TrickplayManager: Creating trickplay files at 320 width, for "/data/media/tv/show/Season 01/S01E01.mkv" [ID: 81beb50e-99bb-d24f-f059-2d4723b3271a]
[2024-05-22 00:15:25.086 -04:00] [INF] [20] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Trickplay generation: "/usr/lib/jellyfin-ffmpeg/ffmpeg -loglevel error -threads 16 -noautorotate -i file:\"/data/media/tv/show/Season 01/S01E01.mkv\" -an -sn -vf \"fps=0.10000000149011612,setparams=color_primaries=bt709:color_trc=bt709:colorspace=bt709,yadif=0:-1:0,scale=trunc(min(max(iw\,ih*a)\,320)/2)*2:trunc(ow/a/2)*2,format=yuv420p\" -threads 16 -c:v mjpeg -qscale:v 4 -f image2 \"/cache/temp/a02bcd3ffff34d7298c330fa534ad9e6/%08d.jpg\""
[2024-05-22 00:21:43.039 -04:00] [INF] [31] Jellyfin.Server.Implementations.Trickplay.TrickplayManager: Finished creation of trickplay files for "/data/media/tv/show/Season 01/S01E01.mkv"
[2024-05-22 00:21:43.073 -04:00] [INF] [31] Emby.Server.Implementations.ScheduledTasks.TaskManager: "Generate Trickplay Images" Completed after 6 minute(s) and 18 seconds
```

### top

```markdown
| UID  | PID    | PPID   | C  | STIME | TTY | TIME     | CMD                                                                                                                                                                                                                                                                                                                                                                                                                     |
|------|--------|--------|----|-------|-----|----------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| root | 104831 | 104810 | 0  | 03:40 | ?   | 00:00:10 | /jellyfin/jellyfin --ffmpeg /usr/lib/jellyfin-ffmpeg/ffmpeg                                                                                                                                                                                                                                                                                                                                                             |
| root | 107687 | 104831 | 99 | 04:14 | ?   | 00:08:45 | /usr/lib/jellyfin-ffmpeg/ffmpeg -loglevel error -threads 16 -noautorotate -i file:/data/media/tv/show/Season 01/S01E01.mkv -an -sn -vf fps=0.10000000149011612,setparams=color_primaries=bt709:color_trc=bt709:colorspace=bt709,yadif=0:-1:0,scale=trunc(min(max(iw\,ih*a)\,320)/2)*2:trunc(ow/a/2)*2,format=yuv420p -threads 16 -c:v mjpeg -qscale:v 4 -f image2 /cache/temp/c0f25652dd4e41948578c18d999a2995/%08d.jpg |
| root | 107849 | 104831 | 99 | 04:15 | ?   | 00:03:55 | /usr/lib/jellyfin-ffmpeg/ffmpeg -loglevel error -threads 16 -noautorotate -i file:/data/media/tv/show/Season 01/S01E01.mkv -an -sn -vf fps=0.10000000149011612,setparams=color_primaries=bt709:color_trc=bt709:colorspace=bt709,yadif=0:-1:0,scale=trunc(min(max(iw\,ih*a)\,320)/2)*2:trunc(ow/a/2)*2,format=yuv420p -threads 16 -c:v mjpeg -qscale:v 4 -f image2 /cache/temp/c9de8a33803048ef98f68d07a80f17a2/%08d.jpg |
| root | 107998 | 104831 | 99 | 04:15 | ?   | 00:00:54 | /usr/lib/jellyfin-ffmpeg/ffmpeg -loglevel error -threads 16 -noautorotate -i file:/data/media/tv/show/Season 01/S01E01.mkv -an -sn -vf fps=0.10000000149011612,setparams=color_primaries=bt709:color_trc=bt709:colorspace=bt709,yadif=0:-1:0,scale=trunc(min(max(iw\,ih*a)\,320)/2)*2:trunc(ow/a/2)*2,format=yuv420p -threads 16 -c:v mjpeg -qscale:v 4 -f image2 /cache/temp/a02bcd3ffff34d7298c330fa534ad9e6/%08d.jpg |
```

### ls /cache/temp

drwxr-xr-x 2 root root 12288 May 22 00:20 c0f25652dd4e41948578c18d999a2995
drwxr-xr-x 2 root root 12288 May 22 00:21 c9de8a33803048ef98f68d07a80f17a2

### FFmpeg logs

_No response_

### Please attach any browser or client logs here

_No response_

### Please attach any screenshots here

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by NotSaifA** [comment]
Found the issue, working on a fix.

**Comment by NotSaifA** [comment]
Merged #11790
